shepherd/shepherd/shepherd.py

```diff
@@ -116,7 +121,25 @@ def enqueue_job(self, job_id: str, job_meta: ModelModel, sheep_id: Optional[str]
         self[sheep_id].jobs_meta[job_id] = job_meta
         self[sheep_id].jobs_queue.put(job_id)
 
-    def health_check(self, sheep_id: str) -> None:
+    def shepherd_health_check(self) -> None:
+        while True:
+            gevent.sleep(1)
+
+            if not self.storage.is_accessible() and not self.storage_inaccessible_reported:
+                logging.warning("The remote storage is not accessible")
+                self.storage_inaccessible_reported = True
+            else:
+                self.storage_inaccessible_reported = False
+
+            if self.registry_config is not None:
+                try:
+                    list_images_in_registry(self.registry_config)
+                    self.registry_inaccessible_reported = False
+                except:
+                    logging.warning("The Docker registry is not accessible")
```

Comment:
This is rather `error` level thing, is it not?

Reply:
It might happen when we're launching a bunch of containers and Minio hasn't started yet, which is an error that will probably go away soon. This led me to a conclusion we don't need to be extra loud about this.

Reply:
On the other hand, it is quite severe if it occurs later on... I would opt for the `error` level...